parser/src/fstring.rs

```diff
@@ -107,14 +107,14 @@ impl<'a> FStringParser<'a> {
                                     formatted_value_piece.push(next);
                                     spec_constructor.push(
                                         self.expr(ExprKind::FormattedValue {
-                                            value: Box::new(
-                                                FStringParser::new(
+                                            value: Box::new(self.expr(ExprKind::JoinedStr {
+                                                values: FStringParser::new(
                                                     &formatted_value_piece,
                                                     Location::default(),
                                                     &self.recurse_lvl + 1,
                                                 )
-                                                .parse()?,
-                                            ),
+                                                    .parse()?
+                                            })),
```

Comment:
This yields a slightly different AST than CPython, we're returning a `FormattedValue(JoinedStr(FormattedValue(` whereas CPython just returns a `FormattedValue`.

Reply:
We thought the return type can be changed, but maybe actually not? I thought The returned JoinedStr is always a vector of strings, but it seems it had a nested structure.

Reply:
Yeah, although the returned `JoinedStr` can either contain `FormattedValue`, `Constant`, or `JoinedStr`, right? I think the code as-written does work, it just includes some additional nesting (but still fewer levels than before IIUC).

Reply:
As written, this does now return `format_spec: Some(JoinedStr(FormattedValue(`, just like CPython. (The `exactly_one` enforcement was necessary, AFAICT.)